include/tvm/topi/nn.h

```diff
@@ -642,6 +643,54 @@ inline tvm::te::Tensor batch_to_space_nd(const tvm::te::Tensor& data,
   out = strided_slice(out, begin_idx, end_idx, strides);
   return out;
 }
+
+/*!
+ * \brief Negative log likelihood loss.
+ *
+ * \param predictions The prediction tensor.
+ * \param targets The target tensor.
+ * \param weights A manual rescaling weight given to each class.
+ * \param reduction The reduction method to apply to the output.
+ * \param ignore_index The target value to ignore.
+ * \param name The name of the operation.
+ * \param tag The tag to mark the operation.
+ *
+ * \return A Tensor whose op member is the batch_to_space_nd operation
+ */
+inline Tensor nll_loss(const Tensor& predictions, const Tensor& targets, const Tensor& weights,
+                       std::string reduction = "mean", int ignore_index = -100,
+                       const std::string name = "nll_loss", const std::string tag = kBroadcast) {
```

Comment:
should the tag be `kOpaque` to match the Relay pattern?

Reply:
@altanh I am confused with the `tag` in topi (ones in `topi/tags.h`) and the `OpPatternKind` in `relay/op_attr_types.h`. It seems that they are not matched. Could you tell me the usage of them in tvm? Thank you~

Reply:
I see, that is confusing.. I'll get back to you on this soon

Reply:
> @altanh Could you take a look at the tests? Thank you~ And I wonder if there is any update with the `tag` and `OpPatternKind`?

I don't have much of an update for the tag, maybe you could try leaving it empty string?